content/posts/irods-development-update-march-2025.md

```diff
@@ -0,0 +1,280 @@
+Title: iRODS Development Update: March 2025
+Date: 2025-03-22 12:00
+Author: Kory Draughn
+Slug: irods-development-update-march-2025
+Status: published
+
+
+Hello reader,
+
+In case you missed it, [iRODS 4.3.4 is available](https://irods.org/2025/03/irods-4-3-4-is-released/)! Check it out and let us know what you think.
+
+_**NOTE:** A bug involving the passing of a `KeyValPair` between the iRODS Rule Language and Python rule engine plugin was discovered following the release of iRODS 4.3.4. A fix will be included in the next version of iRODS. To learn more, see [https://github.com/irods/irods/issues/8265](https://github.com/irods/irods/issues/8265)._
+
+Work on the iRODS 5 server is making good progress. We're cleaning out unused dependencies such as Avro, ZeroMQ, and cppzmq. Work to make the server track access time is making good progress and is on track to land in iRODS 5. For those interested in following the development, see the pull request at [https://github.com/irods/irods/pull/8269](https://github.com/irods/irods/pull/8269).
+
+On the build and packaging side of things, efforts to move more dependencies over to CMake's find_package continues. Not only that, but we're updating CMake policy requirements for newer versions of CMake, looking into a newer version of the Clang compiler, and working on the removal of libarchive as an externals package. See the updates that follow for more information.
+
+The release of iRODS 4.3.4 was a very smooth process. This is due to the many improvements to the server, iCommands, and test infrastructure. However, the release process is still very manual. We are now working on a python script that is capable of testing iRODS across all platforms, hands-free. We're also planning to improve the testing of plugins by simplifying the requirements for running plugin tests, improving concurrency support, and reporting the status of tests while running.
+
+Python iRODS client 3.1.0 is released and ready for use! The big news of this release is that it includes a full port of the iRODS 4.3 authentication framework with working implementations for the native and pam_password authentication schemes. Support for the pam_interactive authentication scheme will be added later. Users looking to use this new version of the client will need Python 3.9 or later.
+
+Notable updates include the Docker-based Development Environment growing support for the Undefined Behavior Sanitizer, work on Metalnx to remove the need for rodsadmin credentials, and efforts to migrate Mungefs to use system-provided packages instead of externals packages.
```

Comment:
```suggestion
Other notable updates include the Docker-based Development Environment growing support for the Undefined Behavior Sanitizer, work on Metalnx to remove the need for rodsadmin credentials, and efforts to migrate Mungefs to use system-provided packages instead of externals packages.
```

Reply:
Done.